### ISSUE TITLE
Bug when `continue` a labeled loop within `coforall`

### ISSUE THREAD

**Issue by Guillaume-Helbecque** [chapel-lang/chapel#21453]
### Summary of Problem
I noted a bug when `continue` a labeled loop within a `coforall` statement. The following code procudes the compile-time error: `error: bad label 'w' on break or continue` while it is not supposed to do (I think).
When using `for` and `forall` loop statements instead of `coforall`, there's no problem.

(In the following code, the use of `label` is not necessary, but it is just a minimal example.)

### Steps to Reproduce

**Source Code:**

```chapel
proc main()
{
  coforall tid in 0..#here.maxTaskPar {
    label w while true {
      continue w; // error
    }
  }
}
```

**Compile command:**
`chpl foo.chpl -o foo.o` 

**Execution command:**
`./foo.o`

### Configuration Information

- Output of `chpl --version`: `1.29.0`
- Output of `$CHPL_HOME/util/printchplenv --anonymize`:
```chapel
CHPL_TARGET_PLATFORM: linux64
CHPL_TARGET_COMPILER: gnu
CHPL_TARGET_ARCH: native *
CHPL_TARGET_CPU: native
CHPL_LOCALE_MODEL: flat
CHPL_COMM: none
CHPL_TASKS: qthreads
CHPL_LAUNCHER: none
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: jemalloc
CHPL_ATOMICS: cstdlib
CHPL_GMP: bundled
CHPL_HWLOC: bundled
CHPL_RE2: bundled
CHPL_LLVM: none *
CHPL_AUX_FILESYS: none
```
- Back-end compiler and version: `gcc 9.4.0`

**Comment by bradcray** [comment]
I've hit this as well, but not had the time to look into it yet.

In my case, I worked around it by just adding more conditional logic within my loop, which was annoying but let me make forward progress.  Please let us know if this is not an option in your code for the time being for some reason (i.e., if this is a blocker).

**Comment by bradcray** [comment]
@Guillaume-Helbecque : I believe I have a fix for this now in #21476.  I need to do more testing on it, and feel certain that there's a cleaner solution, but if you want to see whether the patch works for your copy of the code, I'd be curious what you find.  Otherwise, you can wait until the 1.30 release, which I expect this will make it into in one form or another.

**Comment by Guillaume-Helbecque** [comment]
@bradcray The patch seems to do its job on my copy of the code. Thanks!

**Comment by bradcray** [comment]
Great, thanks for verifying!

**Comment by Guillaume-Helbecque** [comment]
@bradcray I'm facing another related issue using `continue`/`break` within nested `coforall`-statements. Here is the error case using your example above:
```chapel
coforall loc in Locales do on loc {
    coforall tid in 0..#here.maxTaskPar {
      label outer while true {
        var i = 1;
        do {
          writeln(i);
          if i == 3 then
            break outer;
          i += 1;
        } while (i < 100);
      }
    }
  }
```
This code produces the following compile-time error, while the same code without the second `coforall` (tid) is fine.
```
note: the following error is a current limitation
error: break is not allowed in on statement
note: the following error is a current limitation
error: break is not allowed in on statement
error: break is not allowed in coforall statement
error: break is not allowed in coforall statement
```

**Comment by bradcray** [comment]
That is a fascinating error, and one that's completely unfamiliar to me despite being around since 2008.  Let's re-open this to look into it.

**Comment by bradcray** [comment]
@Guillaume-Helbecque : I'm not very proud of it, but https://github.com/chapel-lang/chapel/pull/21528 should squash the errors that you were getting and permits your program above to work.  If you have the chance to try it against your real code that led to this issue, I'd appreciate hearing if that makes things better.

**Comment by Guillaume-Helbecque** [comment]
@bradcray : I confirm that #21528 squashes the error. However, I'd like to point out that this error also occurs using the `continue` statement, and that it would be nice to squash the error in that other case also. In my copy of the code, I simply copy/paste what you did for `break` and that's working.

**Comment by bradcray** [comment]
@Guillaume-Helbecque : This latest comment completely escaped my notice until now.  Thanks for pointing out the symmetrical problem with `continue`.  I've written up the symmetrical fix and will work on getting it in for 1.30.
